### DISCUSSION TITLE
API route returns 413 error (request too large)

### DISCUSSION THREAD

**Discussion by thedavedavies** [vercel/community#2554] [Help] [answered]
### Summary

I'm using NextJS API routes to upload images to S3. Using both localhost and `vercel dev` this works absolutely fine, however when testing on vercel servers I get a 413 error code which indicates that the request was larger than the server is able to handle. Are there limitations on the API?

**Comment by amyegan** [comment, marked as answer]
Hi @thedavedavies. It looks like you're running into the 4.5 MB [Serverless Function Payload Size Limit](https://vercel.com/docs/concepts/limits/overview#serverless-function-payload-size-limit)

**Reply by thedavedavies** [reply]
Ah okay - thanks. Is there better way to upload images to S3, which doesn't hit the payload limit for serverless functions?

**Reply by amyegan** [reply]
You may be able to send larger files by uploading them in the browser, as shown in https://vercel.com/guides/how-can-i-use-aws-s3-with-vercel#uploading-files-in-the-browser
